azavea/raster-foundry
# Allow import into existing project

## Overview

This PR allows a user to begin an export using an existing project.

There are two ways to initiate an upload into an existing project:
- from the project editor, under 'Add/Remove Scenes' there is now an 'Import Scenes' button
- from the project details page, on the right pane, under '+ Add Scenes' there is now '+ Import Scenes'

The flow on the editor seems weird to me but I think tucking the import process within the 'Add/Remove Scenes' would not necessarily be clear to the users.

Other items resolved:
- clicking the 'Remove from project' link no longer opens the detail modal for that scene
- the count of scenes on the project detail page is now correct (previously it was the number of scenes on the current page)
- the 'Share' link in the project dropdown in the editor now works as expected

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo



### Notes

We have a lot of lone `href` attributes in our `a` tags. If there is no `ui-sref` attribute present, this results in the browser thinking the link goes to the baseURL of the application and displaying that url at the bottom of the screen on hover. The UI-Router will automatically add the necessary `href` attributes if they are appropriate so I think we should forgo adding these manually.


## Testing Instructions

 * Make sure the 'Share' option in the project editor dropdown works
 * Make sure the count of scenes in a project on the project details page is correct
 * Make sure that clicking the 'Remove from project' on a scene within the project details page does not open the scene detail modal
 * Click the 'Import Scenes' button within the editor and see that the modal title reflects the fact that we are importing into the current project
 * Do the same for the '+ Import Scenes' link on the project details page

Closes #1677 
Closes #1705
